packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -1195,6 +1195,17 @@ public void onCancel(Object o) {
   }
 
   private void setImageStreamImageAvailableListener(final EventChannel.EventSink imageStreamSink) {
+    // If the imageStreamReader is not initialized yet, initialize it.
```

Comment:
In what situation can `imageStreamReader` be null here?  Shouldn't it be initialized at this point by the `open` function which gets called when the `initialize` platform channel method runs?

Reply:
I don't know why but on some samsung devices this can occur. But it may be because specifically in our product.
For example, our product does the process below:
[Home] --> [Allocate QR Code Reader] --> [Unallocate QR Code Reader] --> [Allocate Camera] --> [Unallocate Camera] --> [Home]
We run this like 400x up to 4000-5000x a day. But 90% of cases happens on some samsung devices.

Reply:
That could be the problem. If you are calling `startImageStream` before intializing the camera controller it may not be ready yet. This is what we do:

```dart
await cameraController.initialize();
await cameraController.startImageStream((CameraImage image) {
  // cameraImageHandle(image);
});
```

Reply:
In my app it's like this:
await cameraController.initialize();
 sleep(Duration(seconds: 1)); 
otherFunction(); <- set some properties
await cameraController.startImageStream((CameraImage image) {
  // cameraImageHandle(image);
});

But thinking about it, I think this could be the reader's communication with the camera and not the camera itself

Reply:
One thing to note. If you are using `sleep` in Dart/Flutter you are sleeping/locking the main thread. If you want a delay it's preferable to do `await Future.delayed(Duration(seconds: 1));`

Reply:
I revert my previous commit and i removed the sleep().